packages/ui/src/theme/types/style-dictionary.d.ts

```diff
@@ -19,8 +19,8 @@ declare module 'style-dictionary/lib/utils/deepExtend.js' {
 declare module 'style-dictionary/lib/utils/flattenProperties.js' {
   // based on the typing specified by style-dictionary:
   // https://github.com/amzn/style-dictionary/blob/main/lib/utils/flattenProperties.js#L16-L24
-  export default function flattenProperties(
+  export default function flattenProperties<T, K = T extends Array<infer TK> ? TK : never>(
```

Comment:
Added the type inference here to remove the reliance on `any`:

![image](https://user-images.githubusercontent.com/15002885/222633924-ef6be008-a444-48fe-963f-7b7334ee28c7.png)